src/utils.ts

```diff
@@ -4,6 +4,10 @@ export const toStars = (addr: string) => {
   if (!addr.startsWith('stars')) {
     const { data } = fromBech32(addr);
     const starsAddr = toBech32('stars', data);
+    // wallet address length 44, contract address length 64, cosmos 20
+    if (![44, 64, 20].includes(data.length)) {
```

Comment:
contract addresses have 64 bytes? Sure? It should be 32 bytes.

Reply:
Ex: `stars12t48c5dvjrghs7jrqaukwlag0jxpt94sgssf3z963u7pvr70039ssss40m` => 64 bytes. They are longer than normal wallet addresses

Reply:
Right, but you are checking the raw data ecoded in the bech32 address here (no prefix, no checksum, no encoding overhead. And this gives you 32 bytes:

```
$ npx @cosmjs/cli@0.28.0-rc1 --code 'import { fromBech32 } from "@cosmjs/encoding"; console.log(fromBech32("stars12t48c5dvjrghs7jrqaukwlag0jxpt94sgssf3z963u7pvr70039ssss40m").data)'
>> Uint8Array(32) [
   82, 234, 124,  81, 172, 144, 209, 120,
  122,  67,   7, 121, 103, 127, 168, 124,
  140,  21, 150, 176,  68,  32, 152, 136,
  186, 143,  60,  22,  15, 207, 124,  75
```

Reply:
44 and 64 still look like bugs to me. I'm only aware of 20 and 32 byte addresses out there.

Reply:
You're right. I didn't realize what this was doing, and now have a greater understanding thanks! `npx @cosmjs/cli@0.28.0-rc1 --code 'import { fromBech32 } from "@cosmjs/encoding"; console.log(fromBech32("stars12t48c5dvjrghs7jrqaukwlag0jxpt94sgssf3z963u7pvr70039ssss40m").data.length)`